tape/src/test/java/com/squareup/tape/QueueFileTest.java

```diff
@@ -640,6 +641,69 @@ public void testForEachReadWithOffset() throws IOException {
     assertThat(iteration.get()).isEqualTo(1);
   }
 
+  @Test public void testScanAll() throws IOException {
+    QueueFile queueFile = new QueueFile(file);
+
+    final byte[] a = { 1, 2 };
+    queueFile.add(a);
+    final byte[] b = { 3, 4, 5 };
+    queueFile.add(b);
+
+    final int[] lengths = queueFile.scanAll();
+    assertThat(lengths.length).isEqualTo(2);
+
+    assertThat(lengths[0]).isEqualTo(2);
+    assertThat(lengths[1]).isEqualTo(3);
+  }
+
+  @Test public void testScanAllOnEmpty() throws IOException {
+    QueueFile queueFile = new QueueFile(file);
+
+    final int[] lengths = queueFile.scanAll();
+    assertThat(lengths.length).isEqualTo(0);
+  }
+
+  @Test public void testScanAllOnWrongLength() throws IOException {
+    QueueFile queueFile = new QueueFile(file);
+    byte[] headerBuffer = new byte[16];
+    writeInts(headerBuffer, QueueFile.INITIAL_LENGTH * 2, 0, 0, 0);
+    queueFile.raf.seek(0);
+    queueFile.raf.write(headerBuffer);
+
+    try {
+      queueFile.scanAll();
+      fail("Should have complained about bad file length!");
+    } catch (IOException ex) {
+      assertThat(ex).hasMessage("File is truncated. Expected length: " + (QueueFile.INITIAL_LENGTH * 2) + ", Actual length: " + queueFile.raf.length());
+    }
+  }
+
+  @Test public void testScanAllOnCorruptedElement() throws IOException {
+    QueueFile queueFile = new QueueFile(file);
```

Comment:
Can you also try with a length larger than 2x the queue size?

Reply:
Yup, will add.